swarm_options.go

```diff
@@ -0,0 +1,19 @@
+package swarm
+
+// Option is a Swarm Option that can be given to a Swarm Constructor(`NewSwarm`).
+type Option func(s *Swarm)
```

Comment:
let's also return an error from the option.

Reply:
the reason I didn't return an error was because if we did, then the
swarm constructor also has to return an error
```go
 NewSwarm(ctx context.Context, local peer.ID, peers peerstore.Peerstore, bwc metrics.Reporter, opts ...Option) (*Swarm,error) {
```
This would break existing code, if you still want me to change, I will 
add it in

Reply:
Ah yes, good point; let's not.

Reply:
or we can panic on error, but that's something we can do from the option itself I guess.